### PR TITLE
Swift 2.3/3.0 Support

### PR DESCRIPTION
**Pull request by dostrander**
- Took the already made branch (thanks @ateliercw) and tweaks and attempts to merge into develop
- Not much work from me, just trying to finish it off and get it out there

### CODE REVIEW

File: Pod/Classes/Deselection/UIViewController+Deselection.swift [hunk 1]
```diff
@@ -30,6 +30,12 @@
 
 import UIKit
 
+#if swift(>=3.0)
+    private typealias IndexPathType = IndexPath
```
Comment (ZevEisenberg) [inline Pod/Classes/Deselection/UIViewController+Deselection.swift]: Why not just `typealias IndexPath = NSIndexPath` on Swift 2.3 only?

Comment (dostrander) [inline Pod/Classes/Deselection/UIViewController+Deselection.swift]: 👍

File: Pod/Classes/Deselection/UIViewController+Deselection.swift [hunk 2]
```diff
@@ -38,27 +44,59 @@ public extension UIViewController {
     ///  controller's `viewWillAppear(_:)` method.
     ///
     ///  - parameter tableView: The table view in which to perform deselection/reselection.
-    func rz_smoothlyDeselectRows(tableView tableView: UITableView?) {
+    func rz_smoothlyDeselectRows(tableView: UITableView?) {
```
Comment (KingOfBrian) [inline Pod/Classes/Deselection/UIViewController+Deselection.swift]: I think `rz_smoothlyDeselectRows(in tableView: UITableView?)` would be a clearer call site with 3.0 naming conventions. Not sure if 3.0 conventions are important at this point.

File: Pod/Classes/Keyboard/Keyboard.swift [hunk 3]
```diff
@@ -63,50 +54,91 @@ final class Keyboard {
...
+     Remove the object as a keyboard frame observer. NOTE: observer is not strongly held, therefore this method is purely optional.
+
+     - parameter observer: The object being observed to remove.
+     */
+    static func removeFrameObserver(observer: AnyObject) {
+    frameObservers.removeObjectForKey(observer)
+
+    if frameObservers.count == 0 {
+    teardownObservers()
+    }
```
Comment (KingOfBrian) [inline Pod/Classes/Keyboard/Keyboard.swift]: Indentation

**Comment by KingOfBrian** [comment]
There is a bunch of inconsistencies with method signatures and Selector name spacing. They should follow one of the two formats:

```
// Use static dispatch and do not create a selector
@nonobjc final func doThing() {}
// Use the specified, namespaced selector:
@objc(rz_doThing) func doThing() {}
```
This seems to be best practice from the swift libraries I've seen. I have not seen any prefixed swift method names!

**Pull request closed without merging**